Replace negz/provider-nop with crossplane/provider-nop

rahulgrover99 opened the pull request:
Signed-off-by: Rahul Grover <rahulgrover99@gmail.com>

<!--
Thank you for helping to improve conformance!

Please read through https://git.io/fj2m9 if this is your first time opening a
conformance pull request. Find us in https://slack.crossplane.io/messages/dev if
you need any help contributing.
-->

### Description of your changes
Since [provider-nop](https://github.com/crossplane-contrib/provider-nop) is ready to use, it might be a good idea to use the same for conformance tests. I have replaced occurrences of `negz/provider-nop` with `crossplane/provider-nop` and added essential spec required for the same. 
<!--
Briefly describe what this pull request does. Be sure to direct your reviewers'
attention to anything that needs special consideration.

We love pull requests that resolve an open conformance issue. If yours does, you
can uncomment the below line to indicate which issue your PR fixes, for example
"Fixes #500":

-->


I have:

- [x] Read and followed conformance's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

I've run the test on a local kind cluster running Crossplane 1.2. All the tests pass successfully except `TestConfiguration`. That might require updating the configuration image being used for the test.

[contribution process]: https://git.io/fj2m9

negz commented on crossplane/apiextensions_test.go:
```diff
@@ -453,13 +453,11 @@ func TestCompositeResource(t *testing.T) {
 		Spec: pkgv1.ProviderSpec{
 			PackageSpec: pkgv1.PackageSpec{
-				Package:                     "negz/provider-nop:v0.1.0",
+				Package:                     "crossplane/provider-nop:main",
```
Should we cut a release of provider-nop we can pin to? I can take care of doing that, since I believe it will require adding some secrets to the provider-nop GitHub repo.

rahulgrover99 replied:
That will be great. Thanks!

negz reviewed (commented):
> All the tests pass successfully except TestConfiguration. That might require updating the configuration image being used for the test.

Could you elaborate on what is failing? Maybe paste the test failure in a comment.

rahulgrover99 commented:
In the `configuration_test.go` file I made this change: 
`prv := &pkgv1.Provider{ObjectMeta: metav1.ObjectMeta{Name: "negz-provider-nop"}}`
to 
`prv := &pkgv1.Provider{ObjectMeta: metav1.ObjectMeta{Name: "crossplane-provider-nop"}}`

which resulted in the following failure: 
```
--- FAIL: TestConfiguration (130.15s)
    /home/rg99/crossplane/conformance/crossplane/configuration_test.go:62: Created configuration "crossplane-conformance"
    --- FAIL: TestConfiguration/DependencyBecomesInstalledAndHealthy (90.02s)
        /home/rg99/crossplane/conformance/crossplane/configuration_test.go:169: Testing that the configuration's dependencies' Healthy and Installed status conditions become 'True'.
        /home/rg99/crossplane/conformance/crossplane/configuration_test.go:192: Provider "crossplane-provider-nop" never became Healthy and Installed: timed out waiting for the condition
    /home/rg99/crossplane/conformance/crossplane/configuration_test.go:76: Cleaning up configuration "crossplane-conformance".
    /home/rg99/crossplane/conformance/crossplane/configuration_test.go:83: Deleted configuration "crossplane-conformance"
    /home/rg99/crossplane/conformance/crossplane/configuration_test.go:65: Cleaning up provider "crossplane-provider-nop".
    /home/rg99/crossplane/conformance/crossplane/configuration_test.go:72: Deleted provider "crossplane-provider-nop"
FAIL
FAIL	github.com/crossplane/conformance/crossplane	130.169s
FAIL
```
